packages/playwright-test/src/common/compilationCache.ts

```diff
@@ -78,10 +78,15 @@ export function getFromCompilationCache(filename: string, code: string, moduleUr
 
   return {
     addToCache: (code: string, map: any) => {
-      fs.mkdirSync(path.dirname(cachePath), { recursive: true });
+      // Make sure we create cache folders with 777 permissions.
+      // Based on the stackoveflow answer: https://stackoverflow.com/a/34721366/314883
+      // See https://github.com/microsoft/playwright/issues/21859
+      const oldmask = process.umask(0);
```

Comment:
Why don't we do this once at the start of the test runner, and not only here? We create quite a few directories.